io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -404,14 +403,13 @@ impl Lvol {
 
         Ok(())
     }
-
     async fn do_create_snapshot(
         &self,
         snap_param: SnapshotParams,
         done_cb: unsafe extern "C" fn(*mut c_void, *mut spdk_lvol, i32),
         done_cb_arg: *mut ::std::os::raw::c_void,
-        receiver: Option<oneshot::Receiver<i32>>,
-    ) -> Result<(), Error> {
+        receiver: oneshot::Receiver<(i32, Lvol)>,
+    ) -> Result<Lvol, Error> {
```

Comment:
Can this function call do_create_snapshot_remote ?

Reply:
no. Currently do_create_snapshot() and do_create_snapshot_remote() are 2 different function for local and remote snapshot operation respectively.